bftclient/include/bftclient/seq_num_generator.h

```diff
@@ -71,9 +72,9 @@ class SeqNumberGenerator {
   // limited to the size lastMilli shifted.
   const uint64_t LAST_COUNT_LIMIT = 0x3FFFFF;
   // lastMilliOfUniqueFetchID_ holds the last SN generated,
-  uint64_t lastMilliOfUniqueFetchID_ = 0;
+  std::atomic<uint64_t> lastMilliOfUniqueFetchID_ = 0;
```

Comment:
If the method `unique` is called from multiple threads, then I am afraid that making `lastMilliOfUniqueFetchID_` and `lastCountOfUniqueFetchID_` atomic does not fix the issue because both variables are tested and changed in multiple places. Making them atomic does not provide logical thread safety.

Reply:
You are right @f-squirrel 
Data race was taken care, however race condition was missed.
Resolved here https://github.com/vmware/concord-bft/pull/1750/commits/92761f7994510c3e65d1b779036ae1aa9b7bfa3d